src/api/delivery-needed/index.js

```diff
@@ -109,3 +122,95 @@ exports.deliveryNeededRequestHandler = async (req, res) => {
     },
   });
 };
+
+/*
+ * This handler takes a delivery request code and the phone number of a delivery volunteer
+ * and assigns the volunteer to the delivery request. It also pings our Twilio service which then
+ * texts the delivery and intake volunteers with relevant information to initiate a
+ * delivery.
+ */
+exports.assignDeliveryHandler = async (req, res) => {
+  const { requestCode, phoneNumber } = req.body;
+
+  if (!requestCode || !phoneNumber)
+    return res.status(400).send({
+      message: "Expected `requestCode` and `phoneNumber` in payload.",
+    });
+
+  let request;
+  let volunteer;
+  let error;
+
+  // validate request
+  try {
+    [request, error] = await findRequestByCode(requestCode);
+    if (error)
+      return res.status(404).send({
+        message: `${requestCode} not found.`,
+      });
+
+    if (request.fields.Status !== status_options.deliveryNeeded)
+      return res.status(400).send({
+        message: `Cannot claim delivery ${requestCode} with status ${request.fields.Status}.`,
+      });
+  } catch (e) {
+    return res.status(400).send({ message: e });
+  }
+
+  // find volunteer record in volunteer table
+  try {
+    [volunteer, error] = await findVolunteerByPhone(phoneNumber);
+
+    // TODO: on not found error, send volunteer form
+    if (!volunteer || (error && error.includes("404")))
+      return res.status(404).send({
+        message: error,
+      });
+    if (error) throw new Error(error);
+  } catch (e) {
+    return res.status(400).send({
+      message: `Something went wrong while looking for volunteer record: ${e}`,
+    });
+  }
+
+  // try {
```

Comment:
is this pending the new webhook from @mab253 ?

Reply:
yes this is- they're going to provide a new endpoint once the webhook is done

Reply:
🎉

here's the new webhook!  it receives: `event.body.requestCode`, and `event.body.phoneNumber`
you can see the code here: [](https://github.com/mab253/mutual-aid-app/blob/master/src/twilio/functions/sms/test-delivery.js) - it looks for records in the **staging base**

and the endpoint that we can use is: https://mutual-aid-4526-dev.twil.io/sms/test-delivery

we ready to test?! @ifearcompilererrors @piratefsh